test/functional/feature_reindex.py

```diff
@@ -68,12 +72,32 @@ def find_block(b, start):
             'LoadExternalBlockFile: Out of order block',
             'LoadExternalBlockFile: Processing out of order child',
         ]):
-            extra_args = [["-reindex"]]
-            self.start_nodes(extra_args)
+            self.start_node(0, ["-reindex"])
 
         # All blocks should be accepted and processed.
         assert_equal(self.nodes[0].getblockcount(), 12)
 
+    def reindex_readonly(self):
+        self.connect_nodes(0, 1)
+        # generate enough blocks to ensure that the -fastprune node fills up the
+        # first blk00000.dat file and starts another block file
+        self.generatetoaddress(self.nodes[1], 300, self.nodes[1].get_deterministic_priv_key().address)
+        self.stop_node(1)
+
+        # make the first block file read-only
+        path = os.path.join(self.nodes[1].chain_path, 'blocks', 'blk00000.dat')
+        os.chmod(path, 0o444)
+
+        # restart and reindex the node with the read-only block file
+        self.start_node(1, ["-reindex", "-fastprune"])
```

Comment:
Verified that without the PR, the node doesn't start, and we see this in `debug.log`:
```
2023-02-04T00:05:32.219184Z [loadblk] [node/blockstorage.cpp:865] [ThreadImport] Reindexing block file blk00001.dat...
2023-02-04T00:05:32.224307Z [loadblk] [flatfile.cpp:44] [Open] Unable to open file /tmp/btest/node1/regtest/blocks/blk00000.dat
2023-02-04T00:05:32.224320Z [loadblk] [util/system.h:50] [error] ERROR: Flush: failed to open file 0
2023-02-04T00:05:32.224334Z [loadblk] [shutdown.cpp:26] [AbortNode] *** Flushing block file to disk failed. This is likely the result of an I/O error.
```
Nothing the PR needs to address, but just FYI, the first message may look confusing; it mentions `blk00001.dat` just before it attempts to write the previous file, `blk00001.dat`, which is where it runs into the write failure. Notice that this write is what the PR prevents because it's unnecessary. This is a nice way to test that the write happens without the PR, and doesn't happen with the PR.